### PR TITLE
warning for using instance method instead of classmethod

### LINKED ISSUE #18169: `load_from_checkpoint` should not be called as an instance method
### Outline & Motivation

Spent a while debugging : 
```python
model.load_from_checkpoint(model_path)
```
which should be : 
```python
model = LitModel.load_from_checkpoint(model_path)
```
there should be a warning when the instance method is called. 

### Pitch

```python
model.load_from_checkpoint(model_path)
```
which should be : 
```python
model = LitModel.load_from_checkpoint(model_path)
```
there should be a warning when the first approach i.e. instance method is called. 

### Additional context

_No response_

cc @justusschock @awaelchli @carmocca @borda

### PR DESCRIPTION
**Pull request by ganessh22**
## What does this PR do?

Fixes #18169 

<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Review by kwsp — COMMENTED** [review]
This doesn't actually do anything. The `_load_from_checkpoint` is called from the class method `load_from_checkpoint`, in which the `cls` argument is guaranteed to be the subclass type (because the [class method is already marked with `@classmethod`](https://github.com/Lightning-AI/lightning/blob/30f42320a5ef4e1cf375f2ccaf1bb127abb4af02/src/lightning/pytorch/core/module.py#L1438C30-L1438C30)). In fact, [`issubclass` raises a `TypeError` if the first argument is an instance and not a type](https://docs.python.org/3.8/library/functions.html#issubclass).

**Comment by kwsp** [comment]
The 2 methods I know of to guarantee a `classmethod` is not called on an instance are 1. override `__getattribute__` and 2. use a custom metaclass. [See this stackoverflow question](https://stackoverflow.com/questions/42322936/how-do-i-disallow-a-classmethod-from-being-called-on-an-instance). Both are pretty involved and hard to maintain, and its up to the lightning maintainers to decide if its worth it.

File: src/lightning/pytorch/core/saving.py [hunk 1]
```diff
@@ -90,6 +90,11 @@ def _load_from_checkpoint(
     # override the hparams with values that were passed in
     checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY].update(kwargs)
 
+    if (
+        isinstance(cls, pl.LightningModule) or isinstance(cls, pl.LightningDataModule):
+        raise ValueError(
+            f"Calling `.load_from_checkpoint` on a model instance is not supported. Please change it to `{type(cls).__name__}.load_from_checkpoint(...)"
+        )
```
Comment (Borda) [inline src/lightning/pytorch/core/saving.py:97]:
```suggestion
    if isinstance(cls, (pl.LightningModule, pl.LightningDataModule)):
        raise ValueError(
            f"Calling `.load_from_checkpoint` on a model instance is not supported. Please change it to `{type(cls).__name__}.load_from_checkpoint(...)"
        )
```

**Comment by carmocca** [comment]
Closing in favor of https://github.com/Lightning-AI/lightning/pull/18432

**Pull request closed without merging**